devdoc/sf_service_config_to_argc_argv_considerations.md

```diff
@@ -0,0 +1,49 @@
+sf_service_config_to_argc_argv_design
+===
+
+References:
+
+[StartServiceA](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicea)
+
+In the context of another project (shall not be named), the application parameters that are passed to the SF application need to be made known to a Windows Service.
+
+Windows Services receive their parameters on the command line when they are started by a call to [StartServiceA].
+
+There are several ways this problem can be approached:

```

Comment:
>several ways [](http://example.com/codeflow?start=10&length=12)

Another consideration: Use some sort of interprocess communication. So process A (the SF one) starts the service (process B) with some minimal arguments for how to do the IPC. Then B can query process A for all the parameters using some RPC, pipes, etc.

Just thinking about a different way that might avoid the concern you called out for B.1

Reply:
Aye. I think there's 2 rather separate ideas here:
1) how to serialize (comes in the next review)
2) how to transport the serialized data and deserialize it in the receiving end.

The easiest way to transport it is through the buildin mechanisms of StartServiceA (argc/argv). If there will be limitations... "when time comes" -> will implement some other IPC.